src/gcode.py

```diff
@@ -253,50 +276,38 @@ def finishLayer():
             args, comment = line.split(";")[:2]
             args = args.split(" ")
             if comment.lower() == "rotation":  # we have either rotation or incline
-                finishLayer()
+                printer.finishLayer()
                 # if any(a.lower().startswith('u') for a in args):  # rotation
-                rotations.append(Rotation(rotations[-1].x_rot, parseRotation(args[1:])))
+                printer.rotations.append(Rotation(printer.rotations[-1].x_rot, parseRotation(args[1:])))
+                printer.currPos.U = printer.rotations[-1].z_rot
             elif comment.lower() == "incline":
-                finishLayer()
+                printer.finishLayer()
                 # if any(a.lower().startswith('v') for a in args):  # incline
-                rotations.append(Rotation(parseRotation(args[1:]), rotations[-1].z_rot))
+                printer.rotations.append(Rotation(parseRotation(args[1:]), printer.rotations[-1].z_rot))
 
-                cone_axis = rotation_matrix([1, 0, 0], np.radians(rotations[-1].x_rot)).dot([0, 0, 1])
+                printer.cone_axis = rotation_matrix([1, 0, 0], np.radians(printer.rotations[-1].x_rot)).dot([0, 0, 1])
+                printer.currPos.V = printer.rotations[-1].x_rot
             elif args[0] == "G0":  # move to (or rotate)
                 pos = args[1:]
 
                 printer.updatePos(pos)
-
-                if len(path) > 1:  # finish path and start new
-                    layer.append(path)
-                x, y, z, a, b = parseArgs(pos, x, y, z, a, b, cone_axis, rotationPoint, abs_pos)
-                path = [Point(x, y, z, a, b)]
             elif args[0] == "G1":  # draw to
                 pos = args[1:]
 
                 printer.updatePos(pos)
-
-                # check U coordinate change
-                if printer.dU == 0:
-                    # U not changed, do noraml arguments parsing
-                    x, y, z, a, b = parseArgs(pos, x, y, z, a, b, cone_axis, rotationPoint, abs_pos)
-                    path.append(Point(x, y, z, a, b))
-                else:
-                    # U changed, get interpolated points
-                    for point in printer.getPoints():
-                        x, y, z, a, b = parseArgs(point, x, y, z, a, b, cone_axis, rotationPoint, abs_pos)
-                        path.append(Point(x, y, z, a, b))
-
             elif args[0] == "G90":  # absolute positioning
-                abs_pos = True
+                printer.abs_pos = True
```

Comment:
as I see from gcode in goosli, we do not generate in relative positioning, so there we can simplify even this

Reply:
I would leave it, the implementation is not so difficult